sdk/confidentialledger/azure-confidentialledger/tests/test_confidential_ledger_client.py

```diff
@@ -0,0 +1,335 @@
+import hashlib
+
+from azure.confidentialledger import (
+    ConfidentialLedgerCertificateCredential,
+    ConfidentialLedgerClient,
+)
+
+from .constants import USER_CERTIFICATE_THUMBPRINT
+from .testcase import ConfidentialLedgerPreparer, ConfidentialLedgerTestCase
+
+
+class ConfidentialLedgerClientTest(ConfidentialLedgerTestCase):
+    def create_confidentialledger_client(self, endpoint, ledger_id, is_aad):
+        self.set_ledger_identity(ledger_id)
+
+        # The ACL instance should already have the potential AAD user added as an Administrator.
+        credential = self.get_credential(ConfidentialLedgerClient)
+        client = self.create_client_from_credential(
+            ConfidentialLedgerClient,
+            credential=credential,
+            ledger_uri=endpoint,
+            # self.network_certificate_path is set via self.set_ledger_identity
+            ledger_certificate_path=self.network_certificate_path,  # type: ignore
+        )
+
+        if not is_aad:
+            # Add the certificate-based user.
+            client.create_or_update_user(
+                USER_CERTIFICATE_THUMBPRINT, {"assignedRole": "Administrator"}
+            )
+
+            credential = ConfidentialLedgerCertificateCredential(
+                certificate_path=self.user_certificate_path
+            )
+            client = ConfidentialLedgerClient(
+                credential=credential,
+                ledger_uri=endpoint,
+                # self.network_certificate_path is set via self.set_ledger_identity
+                ledger_certificate_path=self.network_certificate_path,  # type: ignore
+            )
+
+        return client
+
+    @ConfidentialLedgerPreparer()
+    def test_append_entry_flow_aad_user(self, confidentialledger_endpoint, confidentialledger_id):
+        client = self.create_confidentialledger_client(
+            confidentialledger_endpoint, confidentialledger_id, is_aad=True
+        )
+        self.append_entry_flow_actions(client)
+
+    @ConfidentialLedgerPreparer()
+    def test_append_entry_flow_cert_user(self, confidentialledger_endpoint, confidentialledger_id):
+        client = self.create_confidentialledger_client(
+            confidentialledger_endpoint, confidentialledger_id, is_aad=False
+        )
+        self.append_entry_flow_actions(client)
+
+    def append_entry_flow_actions(self, client):
+        entry_contents = "Test entry from Python SDK"
+        append_result = client.post_ledger_entry({"contents": entry_contents})
+        self.assertTrue(append_result["transactionId"])
```

Comment:
nit: we're trying to remove the `self.assertTrue`, `self.assertEqual` etc calls, ideally you can do `assert append_result["transactionId"]` etc